### DISCUSSION TITLE
Vitest + Rollup error during integration tests in NestJS with an autoloader, export @dec class

### DISCUSSION THREAD

**Discussion by vboechat** [vitest-dev/vitest#6581] [Q&A]
Hey!

I'm trying to use Vitest in a NestJS application and obviously this application has decorators, in my unit tests all tests are working properly, but when I try to run integration tests, it gives me the error
```
Error: Using the export keyword between a decorator and a class is not allowed. Please use `export @dec class` instead.
 ❯ getRollupError node_modules/rollup/dist/es/shared/parseAst.js:392:41
 ❯ convertProgram node_modules/rollup/dist/es/shared/parseAst.js:1070:26
 ❯ parseAstAsync node_modules/rollup/dist/es/shared/parseAst.js:1924:93
 ❯ ssrTransformScript node_modules/vite/dist/node/chunks/dep-CDnG8rE7.js:52319:11
 ❯ loadAndTransform node_modules/vite/dist/node/chunks/dep-CDnG8rE7.js:51917:72
```

I'm using swc.vite() plugin in my vitest.integration.config.mts and it still gives me the error
And this error only happens when I'm using my auto module loader

Does anyone has an idea on how to fix this?

**Comment by trusktr** [comment]
This looks like an issue with Rollup. Is Vitest using Rollup? https://github.com/rollup/rollup

**Reply by trusktr** [reply]
Update! As the problem is with Vitest, I'm assuming that there's a Vite configuration. I found out how to configure Vite so that decorators will work, by setting the `target` options:

- https://github.com/vitejs/vite/issues/18693#issuecomment-2481838134
